packages/notebook-extension/src/index.ts

```diff
@@ -731,6 +733,67 @@ function addCommands(
     },
     isEnabled
   });
+  commands.addCommand(CommandIDs.runInConsole, {
+    label: 'Run Selected Text or Current Line in Console',
+    execute: args => {
+      // Default to not activating the notebook (thereby putting the notebook
+      // into command mode)
+      const current = getCurrent({ activate: false, ...args });
+
+      if (current) {
+        const { context, content } = current;
+
+        let cell = content.activeCell;
+        let path = context.path;
+        // ignore action in non-code cell
+        if (!cell || cell.model.type !== 'code') {
+          return Promise.resolve(void 0);
+        }
+
+        let code = '';
+        const editor = cell.editor;
+        const selection = editor.getSelection();
+        const { start, end } = selection;
+        let selected = start.column !== end.column || start.line !== end.line;
+
+        if (selected) {
+          // Get the selected code from the editor.
+          const start = editor.getOffsetAt(selection.start);
+          const end = editor.getOffsetAt(selection.end);
+          code = editor.model.value.text.substring(start, end);
+        } else {
+          // no selection, submit whole line and advance
+          code = editor.getLine(selection.start.line);
+          const cursor = editor.getCursorPosition();
+          if (cursor.line + 1 !== editor.lineCount) {
+            editor.setCursorPosition({
+              line: cursor.line + 1,
+              column: cursor.column
+            });
+          }
+        }
+        // open a console, create if needed
+        if (code) {
```

Comment:
Same here, I would prefer if this was an early exit, then the primary code path won't have to be nested.

```typescript
if (!code) {
    return Promise.resolve();
}
```